src/procedures/CreateCheckpoint.ts

```diff
@@ -9,8 +9,16 @@ import {
 import { PolymathError } from '../PolymathError';
 import { findEvents } from '../utils';
 import { SecurityToken, Checkpoint } from '../entities';
+import { Factories } from '../Context';
 
-export class CreateCheckpoint extends Procedure<CreateCheckpointProcedureArgs, Checkpoint> {
+export const createRefreshSecurityTokenFactoryResolver = (
+  factories: Factories,
+  securityTokenId: string
+) => async () => {
+  return factories.securityTokenFactory.refresh(securityTokenId);
+};
+
+export class CreateCheckpoint extends Procedure<CreateCheckpointProcedureArgs, Checkpoint | void> {
```

Comment:
Why did you add `void` to the procedure's return type?

Reply:
Due to resolvers. I have removed it I have found the correct way to solve it now